src/core/python/object_v.cpp

```diff
@@ -25,18 +54,32 @@
         T(ScalarTransform3d); T(ScalarTransform4d); T(ScalarMask);             \
     }
 
-/// Implementation detail of mitsuba.core.get_property
+/// Implementation detail of mitsuba.get_property
 #define GET_PROPERTY_T(T)                                                      \
     if (strcmp(type.name(), typeid(T).name()) == 0)                            \
       return py::cast((T *) ptr, py::return_value_policy::reference_internal,  \
                       parent)
 
+/// Implementation detail of mitsuba.set_property
 #define SET_PROPERTY_T(T)                                                      \
     if (strcmp(type.name(), typeid(T).name()) == 0) {                          \
         *((T *) ptr) = py::cast<T>(value);                                     \
         return;                                                                \
     }
 
+/// Macro to iterate over types that can be passed to put_parameter_impl
+#define PUT_PARAMETER_IMPL_T(T)                                                \
+    if (py::isinstance<T>(value)) {                                            \
+        T v = py::cast<T>(value);                                              \
+        self->put_parameter_impl(name, &v, flags, typeid(T));                  \
```

Comment:
I'm trying to understand the necessity of `PublicistTraversalCallback`.

Why can't we use `put_parameter` here? We can template it on `T`, can't we ?

Reply:
We need this because `put_parameter` is a protected method.

Reply:
I think you mean `put_parameter_impl` is protected.

That's exactly my point, `put_parameter` is public so you can just use that instead and avoid the publicist.

Reply:
Sorry for the confusion. The issue here with `put_parameter` is that it required the `typeid` of `value` which is only known at run-time. Hence we need this macro to generate a big switch statement that checks for all the possible types.

If there is another way around (likely more elegant than this) I would be happy to take it!